src/utils/aadUser.ts

```diff
@@ -16,13 +16,33 @@ export const aadUser = {
       },
       responseType: 'json'
     };
-
     const res = await request.get<{ value: { id: string }[] }>(requestOptions);
 
     if (res.value.length === 0) {
       throw Error(`The specified user with user name ${upn} does not exist.`);
     }
 
     return res.value[0].id;
+  },
+
+  /**
+  * Retrieve a single user.
+  * @param mail User E - mail.
+  */
+  async getUserUpnByEmail(email: string): Promise<string> {
```

Comment:
Let's change this function to retrieve the ID of the user instead of the UPN. A user ID is much more reusable than an UPN.